### DISCUSSION TITLE
Dernières corrections sous windows

### DISCUSSION THREAD

**Discussion by Jean-Romain** [Jean-Romain/tabliato#11] [Announcements]
Fils de discussion pour informer des récentes corrections apportée sous Windows. Le lien vers la dernière version disponible sera dans la liste [:link: page de téléchargement](https://jean-romain.github.io/tabliato/download.html)

@Chris-afk71 @agentils @TOTHE34

**Comment by Jean-Romain** [comment]
Dans la version de [aujourd'hui](https://github.com/Jean-Romain/tabliato/issues/10) le rendu audio de la partition marche (d'après mes tests tout du moins). L'archive à télécharger est beaucoup plus lourde (200 Mo) car elle inclue plusieurs soundfont  qui sont des gros fichiers pour faire le rendu audio.

En bas vous avez une liste déroulante pour choisir le rendu audio qui vous plaît le plus. Accordion.sf2 pique un peu les oreilles. Loffet.sf2 est le meilleur je trouve. C'est celui qui est utilisé pour les rendu du [site](https://jean-romain.github.io/tabliato/tablatures.html). Il faut recompiler pour générer un nouveau rendu.

En théorie une fois que vous avez sélectionné un rendu, tabliato doit se ré-ouvrir avec le même rendu selectionné à chaque fois. Et aussi se réouvrir avec le dernier fichier ouvert. Et aussi mémoriser les options d'export. Bref il est sensé mémoriser les réglages. J'essaye de corriger ça à la prochaine version

![Capture d’écran du 2022-03-24 22-39-14](https://user-images.githubusercontent.com/3872279/160044170-72757f55-1bfb-4cdc-86e1-a194627fcb82.png)

**Comment by Jean-Romain** [comment]
Dans la version du 2022-03-25

- La sauvegarde des préférences utilisateurs fonctionne sous Windows
- J'ai retiré les soundfonts pour des raisons de taille de l'archive. Je n'ai gardé que `Accordion.sf2` qui est un fichier léger. A la place il y a un petit bouton pour aller télécharger les soundfonts. Le problème c'est que à chaque fois que vous installez une nouvelle version il faut remettre les soundfonts. Je vais régler ce problème plus tard.
- La page de téléchargement a été mise à jour. Tabliato est maintenant officiellement un logiciel Linux et Windows.

**Comment by Jean-Romain** [comment]
Voici la version 1.0.0 définitive disponible sur la [page de téléchargement](https://jean-romain.github.io/tabliato/download.html). Les prochaines version ne seront plus 1.0.0. 

Les versions Windows et Linux sont sensées être au même niveau maintenant c'est à dire même avec les même fonctionnalités. Tout fonctionne sous Windows normalement.

1. Correction de petits bugs 
   - #19 
   - #20 
   - #21
2. Ajout de quelques fonctionnalités pratiques 
   - #18
3. Sous Windows le téléchargement d'une nouvelle version n'inclue qu'une soundfonts (`Accordion.sf2`) et celles installées par l'utilisateur sont perdue au moment du remplacement du dossier. Ce nest pas pratique mais en attendant d'avoir un système de mise à jour automatique il est possible de re-télécharger automatiquement de nouvelles fonts dans <kbd>Audio</kbd> > <kbd>Installer des fonts sonores</kbd>
![Capture d’écran du 2022-03-27 22-19-11](https://user-images.githubusercontent.com/3872279/160315554-f10c0eda-2003-4023-9ff5-94de09bdb23e.png)
5. Dans <kbd>Aide</kbd> > <kbd>Mise à jour</kbd> il y a un petit utilitaire qui test si une nouvelle version est disponible et renvoie vers la page de téléchargement le cas échéant.
![Capture d’écran du 2022-03-27 22-17-31](https://user-images.githubusercontent.com/3872279/160315373-38d3be1e-c19d-465e-afce-9d7269fc10ce.png)
6. Il est possible de détacher le lecteur audio parce que... pourquoi pas.
7. Corrections et améliorations internes. Notamment quelques détections d'erreurs supplémentaires dans la syntaxe

**Comment by Jean-Romain** [comment]
## Version 1.0.1

Le changement majeur c'est qu'il y a maintenant un vrai installateur windows qui installe proprement tabliato avec icone sur le bureau et dans le menu démarrer. Le truc standard sous windows quoi... Ca va aussi permettre de faire les mises à jour proprement. Quelques correction de bugs aussi (cf changelog).

- [Download](https://jean-romain.github.io/tabliato/download.html) 
- [Changelog](https://github.com/Jean-Romain/tabliato/blob/master/Changelog.md)

Merci pour vos retours sur l'installateur et l'installation Windows. C'est la première fois que je fais ça mais j'ai testé ça me semble marcher bien comme il faut. Maintenant que j'ai fait tout ça, je vais pouvoir me concentrer sur les choses vraiment intéressantes à faire: programmer des fonctionnalités pour produire de belles tablatures.

**Comment by Chris-afk71** [comment]
Ça fonctionne impeccable pour moi l'installateur et tabliato. Merci beaucoup.
Par contre il est sur les onglets "paroles" et "informations complémentaires" de base et même quand je quitte et revient il affiche ses deux onglets là...

**Reply by Jean-Romain** [reply]
Moi aussi ! Insupportable ! Je vais essayer de comprendre ce qui s'est passé. Ça ira à la version1.0.2 demain ou après demain désolé.

**Comment by Jean-Romain** [comment]
## Tabliato 1.0.2

Changement principal, outre la correction du bug rapporté par @Chris-afk71, c'est que tabliato test désormais si il y a une nouvelle version disponible automatiquement à l'ouverture et ouvre une pop-up si c'est le cas. 

Voilà  ! Maintenant je vais me calmer un peu sur les nouvelles versions. On a un truc qui marche pas trop mal, mieux que ça à jamais été en tous cas et je vous remercie pour la motivation que vous m'avez apporté. Maintenant je vais faire des mise à jour moins souvent et uniquement si j'ai des trucs intéressants à ajouter. Et tabliato vous informera de ça automatiquement (mais je vais continuer à poster des message ici).

- [Download](https://jean-romain.github.io/tabliato/download.html) 
- [Changelog](https://github.com/Jean-Romain/tabliato/blob/master/Changelog.md)

**Comment by Jean-Romain** [comment]
## Tabliato 1.0.4

- New: Nouvelle syntaxe simplifiée pour la main gauche. Quand une mesure comporte plusieurs basses et accords différents on peut écrire uniquement les basses et ne pas écrire les accords et laisser tabliato interpréter. Par example en 6/8 avec un motif rythmique "B aB a":
  ```ly
  A >> A aA a
  [C] >> C cC c
  [A a C c] >> A a C c
  [A C] >> A a C c % Nouvelle syntaxe simplifiée
  ```
- Bug: tabliato 1.0.3 ne fonctionnait plus sous Windows à cause de fichiers de configuration modifiés mais non inclus dans la version Windows. Les fichiers de configurations sont maintenant correctement inclus dans la version Windows et les tablatures rendues correctement.

**Comment by Jean-Romain** [comment]
## Tabliato 1.0.5

- Bug: inversion du sens des notes après avoir écrit des accords [#26](https://github.com/Jean-Romain/tabliato/issues/26)
- Bug: barre de tenue des notes qui ne s'arrête pas après un silence [#25](https://github.com/Jean-Romain/tabliato/issues/25)
- Bug: décalage des boutons à deux chiffres avec barre de tenue [#25](https://github.com/Jean-Romain/tabliato/issues/25)
- Bug: crash du logiciel lorsque un silence apparait sans `:` dans un motif p.ex. `\motif [B:4 a:4 r4]`

**Comment by Jean-Romain** [comment]
## Tabliato 1.1.0

- New: support du "point and click" bidirectionnel. Il possible de cliquer sur la tablature pour trouver la ligne de code correspondante et inversement se mettre sur une ligne de code met en evidence la correspondance dans la tablature en surlignant en jaune les éléments.
![linkin](https://user-images.githubusercontent.com/3872279/162634176-aeea1359-b54f-45cb-b77e-f9bfbe4ee02a.gif)
- New: tracking de la musique. Lorsque la pièce est jouée un curseur suit la musique en directe sur le rendu graphique
![music-tracking](https://user-images.githubusercontent.com/3872279/162620281-2843c35a-3855-445d-815b-267fcc24ad7a.gif)
- New: Changement de l'interface pour zoomer et changer de page sur le rendu graphique.
- Change: Les soundfonts sont maintenant téléchargées depuis leur origine sur http://jmi.ovh/DiatonicTab/SoundFonts/
- Bug: nombreux bugs corrigés
